splunk-otel-android/src/test/java/com/splunk/rum/ConnectionUtilTest.java

```diff
@@ -115,4 +121,22 @@ public void quiznos() {
         monitorCaptor.getValue().onLost(null);
         assertEquals(2, notified.get());
     }
+
+    @Test
+    @Config(sdk = {Build.VERSION_CODES.Q, Build.VERSION_CODES.LOLLIPOP})
+    public void shouldNotFailOnImmediateConnectionManagerCall() {
+        NetworkRequest networkRequest = mock(NetworkRequest.class);
+        NetworkDetector networkDetector = mock(NetworkDetector.class);
+        ConnectivityManager connectivityManager = mock(ConnectivityManager.class);
+
+        doAnswer(invocation -> {
+            ConnectivityManager.NetworkCallback callback = invocation.getArgument(1);
+            callback.onAvailable(mock(Network.class));
+            return null;
+        }).when(connectivityManager)
+                .registerNetworkCallback(eq(networkRequest), any(ConnectivityManager.NetworkCallback.class));
```

Comment:
This test doesn't verify the correct behavior on Q though: the `registerDefaultNetworkCallback()` method is called instead of `registerNetworkCallback()`, so this whole stubbing won't execute on Q and the test will always pass.

Reply:
Drat you are correct